### PR TITLE
Create your own enumerable

### PR DESCRIPTION
**Pull request by mckent05**
# In this PR, my partner @hamid-murambiwa  and I have done the following:

- Created a module called `MyEnumeable`
- Added `.all?`, `.any?`, `.filter` methods in the module
- Created a class `MyList` in `list.rb` file and imported `MyEnumerable`.
- Ensured that we have met all project requirements.

### CODE REVIEW

File: list.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require_relative 'enumerable'
+
+class MyList
+  include MyEnumerable
+  def initialize(*list)
+    @list = list
+  end
```
Comment (Dipeshtwis) [inline list.rb]: - [ ] `MyList` class has the `#initialize` method that accepts multiple arguments. I have showed you how you can do that by using your implemented `#each` method.

```ruby
def initialize(*args)
        @list = []
        args.each { |arg| @list << arg }
    end
```

Currently, I am getting this answer after running this command.

Screenshot from 2022-01-21 21-34-51

File: list.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+end
```
Comment (Dipeshtwis) [inline list.rb]: - [ ] I would recommend implementing `#each` method that yields all members of @list in succession. In the project requirements, it is also mentioned. Below is the attached screenshot.

## Project requirement screenshot

Screenshot from 2022-01-21 21-28-21

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @mckent05 & @hamid-murambiwa, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bimbolabuari — APPROVED** [review]
# Project Approved 🚀

Hi Team🥇,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**